diff --git a/proton/vpn/killswitch/backend/linux/wireguard/killswitch_connection_handler.py b/proton/vpn/killswitch/backend/linux/wireguard/killswitch_connection_handler.py
index 3369d9c..7a789ea 100644
--- a/proton/vpn/killswitch/backend/linux/wireguard/killswitch_connection_handler.py
+++ b/proton/vpn/killswitch/backend/linux/wireguard/killswitch_connection_handler.py
@@ -185,7 +185,7 @@ class KillSwitchConnectionHandler:
     async def _run_ip_route_command():
         def run():
             return subprocess.run(  # nosec subprocess_without_shell_equals_true
-                ["/usr/sbin/ip", "route"], capture_output=True, encoding="utf-8", check=True
+                ["ip", "route"], capture_output=True, encoding="utf-8", check=True
             )
 
         loop = asyncio.get_running_loop()
