diff --git a/proton/vpn/app/gtk/util.py b/proton/vpn/app/gtk/util.py
index 33c00ed..aba203e 100644
--- a/proton/vpn/app/gtk/util.py
+++ b/proton/vpn/app/gtk/util.py
@@ -22,7 +22,7 @@ from typing import Callable
 from gi.repository import Gtk
 
 # See: https://docs.gtk.org/gtk4/migrating-3to4.html#set-a-proper-application-id  # pylint: disable=line-too-long # noqa: E501
-APPLICATION_ID = "proton.vpn.app.gtk"
+APPLICATION_ID = "com.protonvpn.www"
 
 
 def connect_once(widget: Gtk.Widget, signal: str, callback: Callable, *args):
