diff --git a/proton/vpn/app/gtk/widgets/main/tray_icon.py b/proton/vpn/app/gtk/widgets/main/tray_icon.py
index db905e2..c798019 100644
--- a/proton/vpn/app/gtk/widgets/main/tray_icon.py
+++ b/proton/vpn/app/gtk/widgets/main/tray_icon.py
@@ -57,7 +57,7 @@ logger = logging.getLogger(__name__)
 
 # Tray icon configuration
 TRAY_TITLE = "Proton VPN"
-TRAY_ICON_NAME = "proton-vpn-sign"
+TRAY_ICON_NAME = "com.protonvpn.www.proton-vpn-sign"
 TRAY_ICON_DESCRIPTION = ""
 
 # StatusNotifierItem specification
diff --git a/proton/vpn/app/gtk/widgets/main/tray_indicator.py b/proton/vpn/app/gtk/widgets/main/tray_indicator.py
index aef7227..f6fac4a 100644
--- a/proton/vpn/app/gtk/widgets/main/tray_indicator.py
+++ b/proton/vpn/app/gtk/widgets/main/tray_indicator.py
@@ -64,19 +64,19 @@ class TrayIndicator:
         see `_on_connection_disconnected` for implementation details.
     """
     DISCONNECTED_ICON = str(
-        ICONS_PATH / f"state-{states.Disconnected.__name__.lower()}.svg"
+        ICONS_PATH / f"com.protonvpn.www.state-{states.Disconnected.__name__.lower()}.svg"
     )
     DISCONNECTED_ICON_DESCRIPTION = str(
         f"VPN {states.Disconnected.__name__.lower()}"
     )
     CONNECTED_ICON = str(
-        ICONS_PATH / f"state-{states.Connected.__name__.lower()}.svg"
+        ICONS_PATH / f"com.protonvpn.www.state-{states.Connected.__name__.lower()}.svg"
     )
     CONNECTED_ICON_DESCRIPTION = str(
         f"VPN {states.Connected.__name__.lower()}"
     )
     ERROR_ICON = str(
-        ICONS_PATH / f"state-{states.Error.__name__.lower()}.svg"
+        ICONS_PATH / f"com.protonvpn.www.state-{states.Error.__name__.lower()}.svg"
     )
     ERROR_ICON_DESCRIPTION = str(
         f"VPN {states.Error.__name__.lower()}"
