diff --git a/proton/vpn/backend/networkmanager/killswitch/wireguard/killswitch_connection_handler.py b/proton/vpn/backend/networkmanager/killswitch/wireguard/killswitch_connection_handler.py
index 5dbe7ae..f91bc0f 100644
--- a/proton/vpn/backend/networkmanager/killswitch/wireguard/killswitch_connection_handler.py
+++ b/proton/vpn/backend/networkmanager/killswitch/wireguard/killswitch_connection_handler.py
@@ -39,7 +39,7 @@ from proton.vpn.backend.networkmanager.killswitch.wireguard.killswitch_connectio
 
 logger = logging.getLogger(__name__)
 
-IP_BINARY_SEARCH_PATH = os.pathsep.join(["/usr/bin", "/usr/sbin", "/sbin", "/bin"])
+IP_BINARY_SEARCH_PATH = os.pathsep.join(["/app/bin", "/usr/bin", "/usr/sbin", "/sbin", "/bin"])
 
 
 @functools.lru_cache(maxsize=1)
